runtime/lua/vim/lsp/util.lua

```diff
@@ -1939,10 +1939,15 @@ end
 
 ---@param window integer|nil: window handle or 0 for current, defaults to current
 ---@param offset_encoding string utf-8|utf-16|utf-32|nil defaults to `offset_encoding` of first client of buffer of `window`
-local function make_position_param(window, offset_encoding)
+---@param row integer|nil 1-indexed row number of position, defaults to current line
+---@param col integer|nil 0-indexed column number of position, defaults to current character
+---@return table position line and character number within buffer
+local function make_position_param(window, offset_encoding, row, col)
```

Comment:
I don't think using 1,0-indexing is a good idea for LSP APIs. see `|api-indexing|`. I think using 0-based indexing for all row and col would be better. The LSP protocol uses 0-based indexing for the Range representations.

The row, col indexing used for this function is only due to `nvim_win_get_cursor` and other mark-related APIs specifically.

Reply:
Gotcha, thanks for the context. I can make this 0 based for input without breaking the existing behavior

Reply:
Actually.... I just realized that `buf.format()` and `make_given_range_params` is using 1,0-indexing 🤔 (which I'd want to change and break but it's difficult to change at this point). Probably wasn't carefully chosen in retrospect or there could be some reason, e.g. wanted to avoid conversion from the callsite; Some other core devs